File: app/src/main/java/com/antyzero/smoksmog/ui/widget/WidgetModule.java
package com.antyzero.smoksmog.ui.widget;


import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public class WidgetModule {

    @Provides
    @Singleton
    public StationWidgetData provideStationWidgetData(Context context) {
        return new StationWidgetData(context);
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/widget/WidgetModule.kt
package com.antyzero.smoksmog.ui.widget


import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Singleton
class WidgetModule {

    @Provides
    @Singleton
    fun provideStationWidgetData(context: Context): StationWidgetData {
        return StationWidgetData(context)
    }
}
